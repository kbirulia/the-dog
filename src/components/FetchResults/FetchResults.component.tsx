import React, { FunctionComponent } from "react";
import ErrorPage from "../ErrorPage";
import Spinner from "../Spinner";
import IErrorStatus from "../../interfaces/errorStatus";

type TFetchResultsProps = {
  error: IErrorStatus | null;
  isLoaded: boolean;
};

const FetchResults: FunctionComponent<TFetchResultsProps> = ({
  error,
  isLoaded,
  children,
}): JSX.Element => {
  if (error) {
    return <ErrorPage title={error.statusText} />;
  }
  if (!isLoaded) {
    return <Spinner />;
  }

  return <>{children}</>;
};

export default FetchResults;
