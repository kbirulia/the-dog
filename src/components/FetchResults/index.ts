export { default } from "./FetchResults.component";
