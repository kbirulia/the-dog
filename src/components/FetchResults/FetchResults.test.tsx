import React from "react";
import renderer from "react-test-renderer";
import FetchResults from "./FetchResults.component";

jest.mock("../Spinner/Spinner.component");
jest.mock("../ErrorPage/ErrorPage.component");

describe("FetchResults", () => {
  test("should render children", () => {
    const tree = renderer
      .create(
        <FetchResults isLoaded error={null}>
          Child
        </FetchResults>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  test("should render error", () => {
    const tree = renderer
      .create(
        <FetchResults isLoaded error={{ status: 404, statusText: "Not found" }}>
          Child
        </FetchResults>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  test("should render spinner", () => {
    const tree = renderer
      .create(
        <FetchResults isLoaded={false} error={null}>
          Child
        </FetchResults>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
