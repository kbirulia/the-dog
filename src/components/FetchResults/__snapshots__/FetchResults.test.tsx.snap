// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FetchResults should render children 1`] = `"Child"`;

exports[`FetchResults should render error 1`] = `"Not found"`;

exports[`FetchResults should render spinner 1`] = `"Spinner"`;
